src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -91,6 +91,74 @@ public class TestExtensions
     /// </summary>
     internal bool AreDataCollectorsCached { get; set; }
 
+    /// <summary>
+    /// Merge two extension maps.
+    /// </summary>
+    /// 
+    /// <param name="extensionMap1">First extension map.</param>
+    /// <param name="extensionMap2">Second extension map.</param>
+    /// 
+    /// <returns>
+    /// A map representing the merger between the two input maps.
+    /// </returns>
+    internal static IDictionary<string, ISet<string>> MergeExtensionMaps(
```

Comment:
Can be reused maybe `MergeMaps` or `Merge` only?

Reply:
I would also avoid prefer getting a new instance of the dictionary with the values merged, rather than getting live reference either to left, right, or new instance. 

The code mixes merging of dictionaries with merging of sets, for probably no measurable impact on performance.

Reply:
Also, the method cannot really be reused since the return type is pretty specific, i.e. `IDictionary<string, ISet<string>>`. We should probably implement a generic merge operation that returns an `IDictionary<K, V>` object, but this seems a little overengineered given the scope.

Reply:
Fixed the return issue. Now we return a new instance representing the result of the merging operation no matter what we do inside the method.